Update gitrepository docs about validation (#9561)
kyma-project/kyma: docs/serverless/06-02-gitrepository-cr.md

This table lists all the possible parameters of a given resource together with their descriptions:

| Parameter | Required for HTTP(s) | Required for SSH | Description |
|-----------|-------------|-------------|---------------|
| **spec.url** | Yes | Yes | Provides the address to the Git repository with the Function's code and dependencies. Depending on whether the repository is public or private and what authentication method is used to access it, the URL must start with the `http(s)`, `git`, or `ssh` prefix, and end with the `.git` suffix. |
| **spec.auth** | No | Yes | Specifies that you must authenticate to the Git repository. |

> | **spec.auth.type** | No | Yes, have to be `key` | Defines if you must authenticate to the repository with a password or token (`basic`), or an SSH key (`key`). This parameter is required if you provide **spec.auth**. |

Author: `This parameter must be set to `key`.` -> in case of using SSH

> | **spec.auth.type** | No | Yes, have to be `key` | Defines if you must authenticate to the repository with a password or token (`basic`), or an SSH key (`key`). This parameter is required if you provide **spec.auth**. |

Author: ```suggestion
| **spec.auth.type** | No | Yes  | Defines if you must authenticate to the repository with a password or token (`basic`), or an SSH key (`key`). This parameter must be set to `key` in case of SSH. |
```
